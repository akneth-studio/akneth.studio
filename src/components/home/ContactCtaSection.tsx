'use client'

import CTAButton from "../CTAButton";

const calendar = process.env.NEXT_PUBLIC_CALENDAR;

export default function ContactCtaSection() {
  return (
    <section className="contact-cta text-center bg-light py-5 px-4">
      <h2>Kontakt</h2>
      <p>
        Masz pytania lub chcesz zlecić projekt? <br />
        Umów się na bezpłatną konsultację lub napisz przez formularz kontaktowy.
      </p>
      <div className="d-flex justify-content-center gap-3 mt-3">
        <CTAButton
          type='button'
          text="Formularz kontaktowy"
          variant="primary"
          to="/contact"
        />
        <CTAButton
          type='button'
          text="Harmonogram spotkań"
          variant="outline-primary"
          onClick={() => window.open(calendar, "_blank")}
        />
      </div>
    </section>
  );
}
