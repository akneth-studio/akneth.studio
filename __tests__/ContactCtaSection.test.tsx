import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import ContactCtaSection from '../src/components/home/ContactCtaSection';
import '@testing-library/jest-dom';

describe('ContactCtaSection component', () => {
  it('renders correctly with heading and buttons', () => {
    render(<ContactCtaSection />);
    expect(screen.getByRole('heading', { name: /kontakt/i })).toBeInTheDocument();
    expect(screen.getByText(/masz pytania lub chcesz zlecić projekt/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /formularz kontaktowy/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /harmonogram spotkań/i })).toBeInTheDocument();
  });

  it('opens calendar link in new tab when Harmonogram spotkań button is clicked', () => {
    window.open = jest.fn();
    render(<ContactCtaSection />);
    const calendarButton = screen.getByRole('button', { name: /harmonogram spotkań/i });
    fireEvent.click(calendarButton);
    expect(window.open).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    const { container } = render(<ContactCtaSection />);
    expect(container).toMatchSnapshot();
  });
});
