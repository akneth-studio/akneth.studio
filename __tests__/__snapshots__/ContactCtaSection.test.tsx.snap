// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ContactCtaSection component matches snapshot 1`] = `
<div>
  <section
    class="contact-cta text-center bg-light py-5 px-4"
  >
    <h2>
      Kontakt
    </h2>
    <p>
      Masz pytania lub chcesz zlecić projekt? 
      <br />
      Umów się na bezpłatną konsultację lub napisz przez formularz kontaktowy.
    </p>
    <div
      class="d-flex justify-content-center gap-3 mt-3"
    >
      <button
        aria-label="Formularz kontaktowy"
        class="btn btn-primary  btn btn-primary"
        data-testid="cta-button"
        role="button"
        tabindex="0"
        type="button"
      >
        Formularz kontaktowy
      </button>
      <button
        aria-label="Harmonogram spotkań"
        class="btn btn-outline-primary  btn btn-primary"
        data-testid="cta-button"
        role="button"
        tabindex="0"
        type="button"
      >
        Harmonogram spotkań
      </button>
    </div>
  </section>
</div>
`;
